packages/ember-htmlbars/lib/helpers/component.js

```diff
@@ -0,0 +1,74 @@
+/**
+@module ember
+@submodule ember-htmlbars
+*/
+import { read } from "ember-metal/streams/utils";
+import { readComponentFactory } from "ember-views/streams/utils";
+import EmberError from "ember-metal/error";
+
+/**
+  `{{component}}` is a `Ember.Handlebars` helper for adding instances of
+  `Ember.Component` to a template. See [Ember.Component](/api/classes/Ember.Component.html)
+   for additional information on how a `Component` functions.
+
+  `{{component}}`'s primary use is for cases where you want to dynamically
+  change which type of component is rendered as the state of your application
+  changes.
+
+  The provided block will be applied as the template for the component.
+
+  Given an empty `<body>` the following template:
+
+  ```handlebars
+  {{! application.hbs }}
+  {{component infographicComponentName}}
+  ```
+
+  And the following application code
+
+  ```javascript
+  App = Ember.Application.create();
+  App.ApplicationController = Ember.Controller.extend({
+    infographicComponentName: function(){
+      if (this.get('isMarketOpen')) {
+        return "live-updating-chart";
+      } else {
+        return "market-close-summary";
+      }
+    }.property('isMarketOpen')
+  });
+  ```
+
+  The `live-updating-chart` component will be appended when `isMarketOpen` is
+  `true`, and the `market-close-summary` component will be appended when
+  `isMarketOpen` is `false`. If the value changes while the app is running,
+  the component will be automatically swapped out accordingly.
+
+  You should not use this helper when you are consistently rendering the same
+  component. In that case, use standard component syntax, for example:
+
+  ```handlebars
+  {{! application.hbs }}
+  {{live-updating-chart}}
+  ```
+
+  @method component
+  @for Ember.Handlebars.helpers
+*/
+export function componentHelper(params, hash, options, env) {
+  Ember.assert(
+    "The `component` helper expects exactly one argument, plus name/property values.",
+    params.length === 1
+  );
+  options.helperName = options.helperName || 'component';
+
+  var componentParam = params[0];
+  var container = this.container || read(this._keywords.view).container;
+  var componentOrStream = readComponentFactory(componentParam, container);
+
+  if (componentOrStream) {
+    env.helpers.view.helperFunction.call(this, [componentOrStream], hash, options, env);
```

Comment:
I think that we would prefer to avoid the "helpers calling helpers" thing, I believe all we need to do is:

```javascript
var props = {
    helperName: options.helperName || 'component'
};
mergeViewBindings(this, props, hash);
appendTemplatedView(this, options.morph, componentOrStream, props);
```

([from here](https://github.com/emberjs/ember.js/blob/master/packages/ember-htmlbars/lib/helpers/view.js#L213-L214))

But we should wait for @mmun and @mixonic to confirm...